fvtest/compilertriltest/SimplifierFoldAbsNegTest.cpp

```diff
@@ -0,0 +1,237 @@
+         int32_t rtn = verifyNode(iter.currentNode());
+         if(rtn)
+            return rtn;
+         }
+      return 0;
+      }
+   protected:
+      virtual int32_t verifyNode(TR::Node *node) = 0;
+   };
+
+/**
+ * Test Fixture for SimplifierFoldAbsNegTest that
+ * selects only the relevant opts for the test case
+ */
+template <typename T>
+class SimplifierFoldAbsNegTest : public TRTest::JitOptTest
+   {
+
+   public:
+   SimplifierFoldAbsNegTest()
+      {
+      /* Add an optimization.
+       * You can add as many optimizations as you need, in order,
+       * using `addOptimization`, or add a group using
+       * `addOptimizations(omrCompilationStrategies[warm])`.
+       * This could also be done in test cases themselves.
+       */
+      addOptimization(OMR::treeSimplification);
+      }
+
+   };
+
+using Types = ::testing::Types<int32_t, int64_t, float, double>;
+TYPED_TEST_CASE(SimplifierFoldAbsNegTest, Types);
+
+
+class NoAbsAbsIlVerifier : public SimplifierFoldAbsNegTestIlVerifierBase
+   {
+   protected:
+   virtual int32_t verifyNode(TR::Node *node) override
+      {
+      if (node->getOpCode().isAbs())
+         {
+         if (node->getChild(0)->getOpCode().isAbs())
+            {
+            return 1;
+            }
+         }
+      return 0;
+      }
+   };
+
+/*
+ * method(T parameter)
+ *   return abs(abs(parameter));
+ */
+TYPED_TEST(SimplifierFoldAbsNegTest, FoldAbsAbs) {
+    char inputTrees[256];
+    std::snprintf(inputTrees, sizeof(inputTrees), "(method return=%s args=[%s]    "
+                                                  " (block                        "
+                                                  "  (%sreturn                    "
+                                                  "   (%sabs                      "
+                                                  "    (%sabs (%sload parm=0))))))",
+                  nameForType<TypeParam>(), nameForType<TypeParam>(),
+                  prefixForType<TypeParam>(), prefixForType<TypeParam>(), prefixForType<TypeParam>(), prefixForType<TypeParam>());
+
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+    NoAbsAbsIlVerifier verifier;
+
+    ASSERT_EQ(0, compiler.compileWithVerifier(&verifier)) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<TypeParam(*)(TypeParam)>();
+
+    // Invoke the compiled method, and assert the output is correct.
+    for (auto test : dataForType<TypeParam>()) {
+        EXPECT_EQ(std::abs(test), entry_point(test));
+    }
+}
+
+
+class NoAbsNegIlVerifier : public SimplifierFoldAbsNegTestIlVerifierBase
+   {
+   protected:
+   virtual int32_t verifyNode(TR::Node *node) override
+      {
+      if (node->getOpCode().isAbs())
+         {
+         if (node->getChild(0)->getOpCode().isNeg())
+            {
+            return 1;
+            }
+         }
+      return 0;
```

Comment:
Couldn't this be simplified to `return node->getOpCode().isAbs() && node->getChild(0)->getOpCode().isNeg()`?

Reply:
Good idea, updated.